### PR TITLE
Retry running npm ci when it fails

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details
Adds an auto retry mechanism to `npm ci` because it is still happening at a somewhat consistent rate (enough to break deploys). This should allow the `npm ci` command to run for 10 minutes, and if it takes longer than 10 minutes or throws an error, it will retry up to 10 times.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/152345

### Tests
1. Verify all checks pass on this PR (including e2e)
2. Verify once merged, we deploy correctly

### CODE REVIEW

File: .github/workflows/desktop.yml [hunk 1]
```diff
@@ -19,7 +19,11 @@ jobs:
                 node-version: '14.x'
 
             - name: Install node packages
-              run: npm ci -ddd
+              uses: nick-invision/retry@7c68161adf97a48beb850a595b8784ec57a98cbb
+              with:
+                timeout_minutes: 10
+                max_attempts: 10
+                command: npm ci
```
Comment (roryabraham) [inline .github/workflows/desktop.yml:26]: did we need `-ddd` ? No idea what that flag does

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ env:
 
 jobs:
     test:
-        runs-on: macos-10.15
+        runs-on: macos-latest
```
Comment (roryabraham) [inline .github/workflows/e2e.yml:12]: I vaguely remember this change being made because e2e tests were failing. I guess we'll see once they finish

**Review by roryabraham — COMMENTED** [review]
LGTM just a few comments/questions. Also ... 9 retries seems like a bit much. Maybe just 3 max attempts?

File: .github/workflows/desktop.yml (hunk 1 above)
Comment (AndrewGable) [reply .github/workflows/desktop.yml:26]: Those are "silly" debug logs, we don't need them anymore with the retry. They were just for investigating

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (AndrewGable) [reply .github/workflows/e2e.yml:12]: I think it was required before we moved to Xcode 12, but now that we are on Xcode 12 we are OK.

**Comment by AndrewGable** [comment]
I'm fine moving to 3, but I'd honestly rather have it just "guaranteed" to succeed. So I picked a big number on purpose.

**Review by roryabraham — APPROVED** [review]

**Pull request merged**